src/Catalog/Registration/RegistrationMakerCatalogItem.cs

```diff
@@ -155,6 +157,43 @@ private Uri GetPackageContentAddress()
             return _packageContentAddress;
         }
 
+        private string GetLicenseUrl()
+        {
+            INode subject = _catalogItem.CreateUriNode(_catalogUri);
+
+            string id = _catalogItem.GetTriplesWithSubjectPredicate(subject, _catalogItem.CreateUriNode(Schema.Predicates.Id)).FirstOrDefault().Object.ToString();
+            string version = _catalogItem.GetTriplesWithSubjectPredicate(subject, _catalogItem.CreateUriNode(Schema.Predicates.Version)).FirstOrDefault().Object.ToString();
+            Triple licenseExpression = _catalogItem.GetTriplesWithSubjectPredicate(subject, _catalogItem.CreateUriNode(Schema.Predicates.LicenseExpression)).FirstOrDefault();
+            Triple licenseFile = _catalogItem.GetTriplesWithSubjectPredicate(subject, _catalogItem.CreateUriNode(Schema.Predicates.LicenseFile)).FirstOrDefault();
+            Triple licenseUrl = _catalogItem.GetTriplesWithSubjectPredicate(subject, _catalogItem.CreateUriNode(Schema.Predicates.LicenseUrl)).FirstOrDefault();
+
+            if (_galleryBaseAddress != null)
+            {
+                if (licenseExpression != null && !string.IsNullOrWhiteSpace(licenseExpression.Object.ToString()))
+                {
+                    return GetOverWriteLicenseUrl(id, version);
+                }
+                if (licenseFile != null && !string.IsNullOrWhiteSpace(licenseFile.Object.ToString()))
+                {
+                    return GetOverWriteLicenseUrl(id, version);
+                }
+            }
+
+            if (licenseUrl != null)
+            {
+                return licenseUrl.Object.ToString();
```

Comment:
Is this internally doing `Uri.ToString()`? If so, this is wrong.
https://faithlife.codes/blog/2010/08/uritostring_must_die/

I vaguely recall "ToString()" working better in later .NET Frameworks -- but I could be wrong. Either way, we should depend on `Uri.AbsoluteUri` if `Uri` is in the picture here.

Reply:
We looked at RDF implementation and they use `Uri.AbsoluteUri`. So we're good.